### ISSUE TITLE
[HOLD #23302] [$500] Assign Task - Subtask changed name incorrectly displayed in LHN

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#27018]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to NewDot
2. Open a chat
3. Navigate to the "Assign Task" section
4. Enter the task title and select any user as the assignee
5. Click "Confirm Task" to create the task
6. In the task report - Click "Assign Task" again to create a sub-task
7. Enter a title for the sub-task and click "Confirm Task"
8. Edit the subtask description and title

## Expected Result:
Subtask changed name should be correctly displayed in LHN

## Actual Result:
Subtask changed name incorrectly displayed in LHN

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.66-2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/6fbef4dd-9f8a-4d8f-ae16-abcdf2643ea7

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c959c0cba48e140d</li>
        <li>Upwork Job ID: 1701885077179801600</li>
        <li>Last Price Increase: 2023-10-04</li>
    </ul>
</details>

**Comment by abdel-h66** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Updates to subtasks (or tasks) names are incorrectly displayed in the LHN

### What is the root cause of that problem?
Editing a task never triggers any updates to the parent reports and does not add the update comments as opposed to the task creation for example where we could see a chat being added to the parent report via 
https://github.com/Expensify/App/blob/5c138163049ef0f036d46c4359c51fcde673066f/src/libs/actions/Task.js#L74-L82

### What changes do you think we should make in order to solve the problem?
We could do the same while editing the task and generate a comment as such in `editTaskAndNavigate`

https://github.com/Expensify/App/blob/5c138163049ef0f036d46c4359c51fcde673066f/src/libs/actions/Task.js#L404 

```
 const optimisticAddCommentReport = ReportUtils.buildOptimisticTaskCommentReportAction(report.reportID, title, taskAssigneeEmail, assigneeAccountID, `Updated a task: ${title}`, parentReportID);
    const lastCommentText = ReportUtils.formatReportLastMessageText(optimisticAddCommentReport.reportAction.message[0].text);
    
    const optimisticParentReport = {
        lastVisibleActionCreated: currentTime,
        lastMessageText: lastCommentText,
        lastActorAccountID: currentUserAccountID,
        lastReadTime: currentTime,
        lastMessageTranslationKey: '',
    };
```

The message displayed must be discussed and see which one fits best, in the example the message will read  `Updated a task: Task name`

### What alternative solutions did you explore? (Optional)
N/A

**Comment by abdel-h66** [comment]
Contributor details
Your Expensify account email: abdelmajid.h66@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01309bb28dda275941

**Comment by JmillsExpensify** [comment]
That's actually a system message that you're pointing to. We've never updated those based on subsequent changes. That said, I do think you're right is relevant for this case, so I'm triaging.

**Comment by jjcoffee** [comment]
@abdel-h66's [proposal](https://github.com/Expensify/App/issues/27018#issuecomment-1712192757) has the correct RCA that parent reports are not being updated (and indeed this happens for all tasks, not only sub-tasks!), however this is also a BE issue as when we call `EditTask`, `lastMessageText` is not updated for the parent report in the response. I believe we should apply that BE change first and then proceed with adding the `optimisticData` on FE.

:ribbon::eyes::ribbon: C+ reviewed

**Comment by NikkiWines** [comment]
Agreed with @jjcoffee - I can spin up the PR to add `lastMessageText` on the back end and then once that's deployed we can assign this issue to @abdel-h66

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by NikkiWines** [comment]
Going to try and reproduce this one more since QA hasn't been able to

**Comment by jjcoffee** [comment]
@NikkiWines If it helps, I can repro in latest staging (v1.3.75-8). Maybe QA are missing that it's a sub-task? :sweat_smile:

**Comment by NikkiWines** [comment]
Sorry for the delay on this - had some competing priorities of higher value. Confirmed I can reproduce this still and looked into this yesterday. 

On the backend, it's a little more complex than just returning the `lastMessageText` but will try to spin something up today or early next week. Thanks for your patience! 🙇

**Comment by JmillsExpensify** [comment]
Thanks for testing @NikkiWines. Coming back to this issue, it's still internal, and we're working through it as time permits.

**Comment by marcaaron** [comment]
Grabbing this one from @NikkiWines. Will try to get a fix up asap.

**Comment by marcaaron** [comment]
After reading through the issue description... 

I am noticing that when creating a task there is no “system message”, just a task. I cannot see the “Created a task” or the “Assigned a task to you” messages when testing locally.

When a task is created all I see is: 

- `task for [Task title]`

Before we start here... is that the correct behavior? Or did something regress somewhere since this issue was created? @thienlnam might know.

<img width="994" alt="2023-10-13_10-14-12" src="https://github.com/Expensify/App/assets/32969087/05e640bc-ad94-4257-ae3b-5437346ad79b">

**Comment by marcaaron** [comment]
I guess separately - I am seeing that the "Subtask" is also underneath the Main Task on the main chat report. Is that consistent with our expectations of how this feature should work? I would maybe expect the "Subtask" to be inside the "Main Task", but unsure if we designed this in a way where sub-tasks show on the main chat where they are shared.

**Comment by thienlnam** [comment]
> Before we start here... is that the correct behavior?

Yeah, we've updated the 'system messages' so that they're more consistent with our money request flows. 


> I guess separately - I am seeing that the "Subtask" is also underneath the Main Task on the main chat report. Is that consistent with our expectations of how this feature should work? I would maybe expect the "Subtask" to be inside the "Main Task", but unsure if we designed this in a way where sub-tasks show on the main chat where they are shared.

Ah yes, so the "subtask" is indeed a child on the main task but the thing that you're seeing in the DM is just a link to the new task that was assigned to you. 

So if you click on the main task you should indeed see the subtask there, but if you were assigned a task you get a link to it in the DM between you and the task creator (this is for the situation where you may not have access to the main task)

**Comment by marcaaron** [comment]
Cool - thanks! 

So, I guess given that we have made the system messages align more with the money request flows. I'm still a little confused about the proposed solution here. It sounds like we landed on **adding another system message saying that the task title was updated**.

But before going down that road - I'm curious - do we really need it?

Checking out the current behavior and this is the only thing that I see wrong in the before and after when changing a "subtask".

**Before**
![2023-10-13_14-57-20](https://github.com/Expensify/App/assets/32969087/3b39ab87-37d9-4992-a371-2779db144c0b)

**After editing...**
![2023-10-13_14-58-09](https://github.com/Expensify/App/assets/32969087/768bd5b8-2f8b-4b69-984a-5d81af8dca0e)

Refreshing does not fix that. Hmm that's weird. Well, the data must be wrong. 

Looking at what got changed when we edit a task and I see that the report action still is wrong:

```
created                  reportID          accountID  action         message                                                                                                                   reportActionID
-----------------------  ----------------  ---------  -------------  ------------------------------------------------------------------------------------------------------------------------  -------------------
2023-10-14 00:57:54.201  8965468924874847  17106      TASKEDITED     {"assignee":"tasker@task.com","description":"","lastModified":"2023-10-14 00:57:54.201","title":"One - A"}                8295811169164450580
```

Seems weird because the task report itself looks correct. So I looked more and we are also using the `reportName` field to store the task title + description.

What is the solution here? Is it enough to just update the title of this reportAction??

**Comment by JmillsExpensify** [comment]
Wow great sleuthing. I think I agree with @marcaaron that we'd just update the title of the reportAction. Equally, I agree that I'm not sure that we need a system message for updating the report title. We actually don't have a system message for that anywhere yet.

**Comment by marcaaron** [comment]
@thienlnam can you give any insight about whether the title in the report action should be updated? Or should we just put this one on HOLD for now? Not sure what action if any should be taken here.

**Comment by thienlnam** [comment]
I think we can just do nothing for this issue right now - it's already kind of expected that the title in the reportAction doesn't update because it's a stale reportAction (also, because you can change the assignees an infinite number of times we'd have to find all the old reportActions and update them at once). Once we add the edit system messages it will also help address this issue. So I don't really feel that strongly to do anything for this right now though if anyone feels otherwise we can do it

**Comment by marcaaron** [comment]
I agree with putting it on hold for now.

**Comment by JmillsExpensify** [comment]
Let's keep it open for now, though on HOLD. Sounds good to me.

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Same

**Comment by marcaaron** [comment]
Maybe this one can get moved into #vip-vsb

**Comment by JmillsExpensify** [comment]
@thienlnam Do you consider this part of "finish tasks"?

**Comment by thienlnam** [comment]
I think we can close this from my original comment 

https://github.com/Expensify/App/issues/27018#issuecomment-1765430483

As part of "finish tasks" we're adding edit system messages so this issue is no longer a problem

**Comment by marcaaron** [comment]
Sounds good. Let's 🔪
